ckanext/ogdchcommands/commands.py

```diff
@@ -470,3 +493,79 @@ def _print_shacl_validation_result(self, data_dict):
         print(dg.sort_values('count', ascending=False)
               .to_string(index=False))
         print("\n")
+
+
+
+    def cleanup_harvestsources(self, source=None):
+        """
+        command for the harvester job that did not run for > 1 month
+        to cleanup their sources and datasets
+        :argument tf_to_keep_harvested_dsets: int (optional)
+        """
+        # get source from arguments
+        source_id = None
+        data_dict = {}
+        if len(self.args) >= 2:
+            source_id = unicode(self.args[1])
+            data_dict['harvest_source_id'] = source_id
+            print('cleaning up jobs for harvest source {}'.format(source_id))
+        else:
+            print('cleaning up jobs for all harvest sources')
+
+        # get named arguments
+        data_dict['timeframe_to_keep_harvested_datasets'] = self.options.tf_to_keep_harvested_dsets
+
+        # set context
+        context = {'model': model,
+                   'session': model.Session,
+                   'ignore_auth': True}
+        admin_user = logic.get_action('get_site_user')(context, {})
+        context['user'] = admin_user['name']
+
+        # test authorization
+        try:
+            logic.check_access('harvest_sources_clear', context, data_dict)
+            print("User is authorized to perform this action.")
+        except logic.NotAuthorized:
+            print("User is not authorized to perform this action.")
+            sys.exit(1)
+
+        print(data_dict)
+
+        # gets all active harvest sources
+        harvest_sources = model.Session.query(HarvestSource).all()
```

Comment:
check if `harvest_source_show` can work as well.

Reply:
Use ```harvest_sources = tk.get_action('harvest_source_list')(context, data_dict)```
instead of ```model.Session```